Lavinia-api/Controllers/ApiController.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LaviniaApi.Data;
+using LaviniaApi.Models;
+using LaviniaApi.Utilities;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace LaviniaApi.Controllers
+{
+    /// <inheritdoc />
+    /// <summary>
+    ///     Controller that serves data regarding political elections in various levels, for the Mandater project at the
+    ///     University of Oslo, department of political science
+    /// </summary>
+    [EnableCors("CorsPolicy")]
+    [Produces("application/json")]
+    [Route("api/v2.0.0/")]
+    public class ApiController : Controller
+    {
+        private readonly ElectionContext _context;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        ///     Constructor for the ApiController, enables database access and logging
+        /// </summary>
+        /// <param name="context">ElectionContext object that allows access to the database</param>
+        /// <param name="logger">Logger that gives information about the context of a log message</param>
+        public ApiController(ElectionContext context, ILogger<ApiController> logger)
+        {
+            _context = context;
+            _logger = logger;
+        }
+
+
+        /// <summary>
+        ///     Returns a list of all Party votes that meet the required parameters.
+        /// </summary>
+        /// <param name="year">Four digit election year</param>
+        /// <param name="partyCode">One to N character party code</param>
+        /// <param name="district">Name of district</param>
+        /// <returns>Party votes meeting the requirements</returns>
+        [ProducesResponseType(typeof(IEnumerable<PartyVotes>), 200)]
+        [ProducesResponseType(500)]
+        [HttpGet("votes")]
+        public IActionResult GetVotes(int year = 0, string partyCode = "ALL", string district = "ALL")
+        {
+            _logger.LogInformation("GetVotes called with parameters year = " + year + ", partyCode = " + partyCode + ", district = " + district);
+            try
```

Comment:
This whole block is very hard to read, and the length of the Linq query should definitely not extend to 190+ col. This applies to several of the methods. Constants for defaults should be used instead of manual calls such as "ALL".

I recommend following the type? approach and checking for undefined types instead for the .Where statement. [Null propagation](https://msdn.microsoft.com/en-us/magazine/dn802602.aspx) and the convenient short-hand if else operator seems like a natural thing to use here for better readability